### PR TITLE
Avatar: emit error event when loading img fails

### LINKED ISSUE #3419: [Enhancement] Kirby Avatar should handle broken links
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the enhancement
Kirby avatar component should handle broken link for attribute [imageSrc] with an error handle event

## Describe the solution you'd like
Kirby avatar component is expanded with a onerror event handler to handle broken links in [imageSrc].
Therefor we would like the kirby avatar component to be expanded with an error event handler.

## Are there any additional context?

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#the-process-of-contributing) correctly.

### Refinement

- [ ] Request that the issue is [UX refined](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#ux-refinement)
- [ ] Request that the issue is [tech refined](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#tech-refinement); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](../CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [develop branch](https://github.com/kirbydesign/designsystem/tree/develop) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Publish a WIP implementation to Github as a draft PR and ask for feedback. 
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Update the [cookbook](https://cookbook.kirby.design) with examples and showcases.

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request. If you created a draft PR during implementation you can just mark that as "ready for review".

### PR DESCRIPTION
**Pull request by Fuzzy3**
## Which issue does this PR close?

This PR closes #3419 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#the-process-of-contributing) correctly.

### Reminders
- [x] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [x] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [x] Determine if your changes are a fix, feature or breaking-change, and add the matching label to your PR. If it is tooling, dependency updates or similar, add ignore-for-release.
- [x] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [x] Request that the changes are code-reviewed 
- [x] ~~Request that the changes are [UX reviewed] (https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#ux-review) (only necessary if your PR introduces visual changes)~~

When the pull request has been approved it will be merged to `develop` by Team Kirby.

### CODE REVIEW

**Review by jakobe — DISMISSED** [review]
👌🏻

**Review by jakobe — DISMISSED** [review]

File: apps/cookbook/src/app/examples/avatar-example/examples/image-error.ts [hunk 1]
```diff
@@ -2,18 +2,34 @@ import { Component } from '@angular/core';
...
 export class AvatarExampleImageErrorComponent {
   template: string = config.template;
+  ts: string = config.ts;
+  avatarSrc: string = 'bad-image-url.png';
 
   showFallbackImageOnError(event: ErrorEvent) {
-    (event.target as HTMLImageElement).src = '/assets/images/avatar-not-found.png';
+    const fallbackImageSrc = 'assets/images/avatar-not-found.png';
+    this.avatarSrc = fallbackImageSrc;
+    // Possibly do something with the ErrorEvent...
```
Comment (jakobe) [inline apps/cookbook/src/app/examples/avatar-example/examples/image-error.ts]:
```suggestion
    // Possibly do something with the ErrorEvent parameter...
```

File: apps/cookbook/src/app/showcase/avatar-showcase/avatar-showcase.component.html [hunk 2]
```diff
@@ -32,8 +32,22 @@ <h2>Avatar with image</h2>
     <cookbook-avatar-example-image-size #imageSizeExample></cookbook-avatar-example-image-size>
   </cookbook-example-viewer>
 
+  <h2>Avatar with fallback image</h2>
+  <p>
+    The image avatar will emit an
+    <code>ErrorEvent</code>
+    if the image failed to load. This allows for falling back to a default avatar image or do some
```
Comment (jakobe) [inline apps/cookbook/src/app/showcase/avatar-showcase/avatar-showcase.component.html]:
```suggestion
    if the image fails to load. This allows for falling back to a default avatar image or do some
```

File: apps/cookbook/src/app/showcase/avatar-showcase/avatar-showcase.component.html [hunk 3]
```diff
@@ -32,8 +32,22 @@ <h2>Avatar with image</h2>
...
+    other handling when the image could not be loaded.
```
Comment (jakobe) [inline apps/cookbook/src/app/showcase/avatar-showcase/avatar-showcase.component.html]:
```suggestion
    other handling when the image cannot be loaded.
```

File: apps/cookbook/src/app/examples/avatar-example/examples/image-error.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  (imageError)="showFallbackImageOnError($event)"
+  size="lg"
+></kirby-avatar>`,
+  ts: `avatarSrc: string = 'bad-image-url.png';
+
+showFallbackImageOnError(event: ErrorEvent) {
+  const fallbackImageSrc =
+                'assets/images/avatar-not-found.png';
+  this.avatarSrc = fallbackImageSrc;
+  // Possibly do something with the ErrorEvent...
```
Comment (Fuzzy3) [inline apps/cookbook/src/app/examples/avatar-example/examples/image-error.ts]:
```suggestion
  // Possibly do something with the ErrorEvent parameter...
```

File: apps/cookbook/src/app/examples/avatar-example/examples/image-error.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+></kirby-avatar>`,
+  ts: `avatarSrc: string = 'bad-image-url.png';
+
+showFallbackImageOnError(event: ErrorEvent) {
+  const fallbackImageSrc =
+                'assets/images/avatar-not-found.png';
+  this.avatarSrc = fallbackImageSrc;
+  // Possibly do something with the ErrorEvent parameter...
+}
+  `,
```
Comment (jakobe) [inline apps/cookbook/src/app/examples/avatar-example/examples/image-error.ts]: Trailing linebreaks will show up in the code snippet as it's pre-formatted:
image
```suggestion
}`,
```

**Review by jakobe — DISMISSED** [review]
Nice work 👌🏻

**Review by jakobe — APPROVED** [review]

**Pull request merged**